engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -519,28 +520,77 @@ public static FileSystem newFileSystemInstance(Configuration conf) throws IOExce
     }
   }
 
+  /**
+   * Helper function used by both getApexDFSRootDir and getDTDFSRootDir to process dfsRootDir as per the required logic
+   *
+   * @param fs
+   * @param conf
+   * @param dfsRootDir
+   * @param useImpersonated
+   * @param processRelativePath
+   * @return
+   */
+  private static Path processDFSRootDirHelper(FileSystem fs, Configuration conf, String dfsRootDir, boolean useImpersonated,
```

Comment:
Do you want to call it eval instead of process for both method name and the parameter.